homeassistant/components/elkm1/__init__.py

```diff
@@ -175,14 +233,16 @@ def __init__(self, element, elk, elk_data):
         """Initialize the base of all Elk devices."""
         self._elk = elk
         self._element = element
+        self._prefix = elk_data['prefix']
         self._temperature_unit = elk_data['config']['temperature_unit']
-        self._unique_id = 'elkm1_{}'.format(
-            self._element.default_name('_').lower())
+        self._unique_id = 'elkm1_{mac}_{name}'.format(
+            mac=elk.mac_address,
```

Comment:
If the `mac_address` is None` the unique_id won't be unique. Then we have to return `None` for the unique_id.